src/main/scala/com/gu/invoicing/refundErroneousPayment/Program.scala

```diff
@@ -3,16 +3,33 @@ package com.gu.invoicing.refundErroneousPayment
 import com.gu.invoicing.refundErroneousPayment.Impl._
 import com.gu.invoicing.refundErroneousPayment.Model._
 
+import java.time.LocalDate
+
 object Program {
 
   def program(input: RefundInput): RefundOutput = {
-    val RefundInput(accountId, invoiceNumber, paymentId, invoiceAmount, comment, _) = input
-    val balancingInvoiceNumber = getBalancingInvoice(accountId, invoiceAmount)
-    transferToCreditBalance(balancingInvoiceNumber, invoiceAmount, comment)
-    val refundId = createRefundObject(invoiceAmount, paymentId, comment)
-    assert(getRefundStatus(refundId) == "Processed")
-    applyCreditBalance(invoiceNumber, invoiceAmount, comment)
-    assert(getAccountBalance(accountId) == 0)
-    RefundOutput(accountId, paymentId, invoiceAmount, invoiceNumber, balancingInvoiceNumber)
+    val RefundInput(accountId, paymentDate, comment, _) = input
+    val payments = getPayments(accountId, paymentDate)
+    assert(payments.nonEmpty, "No payments")
+    val invoices = getInvoices(accountId)
+    assert(invoices.nonEmpty, "No invoices")
+    val balancingInvoice = invoices.maxBy(_.InvoiceDate)
```

Comment:
in some cases (I found 4), the balancing invoice was created on the same date as the actual invoice.  i.e. a charge was attempted, they got a PF email, and they phoned to cancel straight away

Reply:
Fixed that now